macros/src/actix.rs

```diff
@@ -1069,6 +1061,210 @@ pub fn emit_v2_security(input: TokenStream) -> TokenStream {
+                                }
+                            } else {
+                                emit_warning!(span, invalid_attr_msg);
+                            }
+                        } else {
+                            emit_error!(span, invalid_attr_msg);
+                        }
+                    }
+                    _ => {
+                        emit_error!(span, invalid_attr_msg);
+                    }
+                }
+            }
+        }
+
+        if parameter_attrs.contains_key("skip") {
+            continue;
+        }
+
+        let docs = (!docs.is_empty()).then(|| docs.to_owned());
+        let quoted_description = quote_option(parameter_attrs.get("description").or(docs.as_ref()));
+        let name_string = field_name.as_ref().map(|name| name.to_string());
+        let quoted_name = if let Some(name) = parameter_attrs.get("name").or(name_string.as_ref()) {
+            name
+        } else {
+            emit_error!(
+                field.span(),
+                "Missing header name. Either add a name using the openapi attribute or use named struct parameter"
+            );
+            return quote!().into();
+        };
+
+        let (quoted_type, quoted_format) = if let Some(ty_ref) = get_field_type(field) {
+            (
+                quote! { {
+                    use paperclip::v2::schema::TypedData;
+                    Some(#ty_ref::data_type())
+                } },
+                quote! { {
+                    use paperclip::v2::schema::TypedData;
+                    #ty_ref::format()
+                } },
+            )
+        } else {
+            (quote! { None }, quote! { None })
+        };
+
+        let (quoted_type, quoted_format) = if let Some(format) = parameter_attrs.get("format") {
+            let quoted_format = quote_format(format.clone());
+            let quoted_type = quote! { #quoted_format.map(|format| format.into()) };
+            (quoted_type, quoted_format)
+        } else {
+            (quoted_type, quoted_format)
+        };
+
+        let def_block = quote! {
+            paperclip::v2::models::Parameter::<paperclip::v2::models::DefaultSchemaRaw> {
+                name: #quoted_name.to_owned(),
+                in_: paperclip::v2::models::ParameterIn::Header,
+                description: #quoted_description,
+                data_type: #quoted_type,
+                format: #quoted_format,
+                required: Self::required(),
+                ..Default::default()
+            }
+        };
+
+        header_definitions.push(def_block);
+    }
+
+    let gen = quote! {
+        impl #impl_generics paperclip::v2::schema::Apiv2Schema for #name #ty_generics #where_clause {
+            fn header_parameter_schema() -> Vec<paperclip::v2::models::Parameter<paperclip::v2::models::DefaultSchemaRaw>> {
+                let mut headers = vec![];
+                #(headers.push(#header_definitions));*;
+                headers
```

Comment:
Could we have something like this:
```suggestion
                vec![
                    #(#header_definitions),*
                ]
```